test/framework/integration_framework/fake_peer/fake_peer.hpp

```diff
@@ -49,157 +49,158 @@ namespace iroha {
+              &common_objects_factory,
+          std::shared_ptr<TransportFactoryType> transaction_factory,
+          std::shared_ptr<shared_model::interface::TransactionBatchParser>
+              batch_parser,
+          std::shared_ptr<shared_model::interface::TransactionBatchFactory>
+              transaction_batch_factory);
+
+      /// Assign the given behaviour to this fake peer.
+      FakePeer &setBehaviour(const std::shared_ptr<Behaviour> &behaviour);
+
+      /// Start the fake peer.
+      void run();
+
+      /// Get the address:port string of this peer.
+      std::string getAddress() const;
+
+      /// Get the keypair of this peer.
+      const shared_model::crypto::Keypair &getKeypair() const;
+
+      rxcpp::observable<MstMessagePtr> get_mst_states_observable();
+
+      /// Get the observable of YAC states received by this peer.
+      rxcpp::observable<YacMessagePtr> get_yac_states_observable();
+
+      /// Get the observable of OS batches received by this peer.
+      rxcpp::observable<OsBatchPtr> get_os_batches_observable();
+
+      /// Get the observable of OG proposals received by this peer.
+      rxcpp::observable<OgProposalPtr> get_og_proposals_observable();
+
+      /**
+       * Send the real peer votes from this peer analogous to the provided ones.
+       *
+       * @param incoming_votes - the votes to take as the base.
+       */
+      void voteForTheSame(const YacMessagePtr &incoming_votes);
+
+      /**
+       * Make a signature of the provided hash.
+       *
+       * @param hash - the hash to sign
+       */
+      std::shared_ptr<shared_model::interface::Signature> makeSignature(
+          const shared_model::crypto::Blob &hash) const;
+
+      /// Make a vote from this peer for the provided YAC hash.
+      iroha::consensus::yac::VoteMessage makeVote(
+          const iroha::consensus::yac::YacHash &yac_hash);
+
+      /// Send the main peer the given YAC state.
+      void sendYacState(
+          const std::vector<iroha::consensus::yac::VoteMessage> &state);
+
+      void sendProposal(
+          std::unique_ptr<shared_model::interface::Proposal> proposal);
+
+      void sendBatch(
+          const std::shared_ptr<shared_model::interface::TransactionBatch>
+              &batch);
+
+     private:
+      using MstTransport = iroha::network::MstTransportGrpc;
+      using YacTransport = iroha::consensus::yac::NetworkImpl;
+      using OsTransport = iroha::ordering::OrderingServiceTransportGrpc;
+      using OgTransport = iroha::ordering::OrderingGateTransportGrpc;
+      using AsyncCall =
+          iroha::network::AsyncGrpcClient<google::protobuf::Empty>;
+
+      std::shared_ptr<shared_model::interface::CommonObjectsFactory>
+          common_objects_factory_;
+
+      const std::string listen_ip_;
+      size_t internal_port_;
+      std::unique_ptr<shared_model::crypto::Keypair> keypair_;
+
+      std::shared_ptr<shared_model::interface::Peer>
+          this_peer_;  ///< this fake instance
```

Comment:
```suggestion
          this_peer_;  /// fake instance
```

Reply:
nope https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html#memberdoc

Reply:
`this` still seems redundant

Reply:
It specifies what peer is referred to. See the next declaration, it contains another peer reference.